core/tee/tee_pobj.c

```diff
@@ -64,14 +65,37 @@ TEE_Result tee_pobj_get(TEE_UUID *uuid, void *obj_id, uint32_t obj_id_len,
 {
 	TEE_Result res = TEE_SUCCESS;
 	struct tee_pobj *o = NULL;
+	void *obj_id_kbuf = NULL;
+	bool obj_id_from_user = true;
 
 	*obj = NULL;
 
+	res = memdup_user(obj_id, obj_id_len, &obj_id_kbuf);
+	switch (res) {
+	case TEE_SUCCESS:
+		obj_id_from_user = true;
+		break;
+	case TEE_ERROR_ACCESS_DENIED:
+		/*
+		 * Since the caller of tee_pobj_get() checks if obj_id is
+		 * accessible if obj_id is a user address
+		 * (See vm_check_access_rights), we know that if
+		 * copy_from_user(..., obj_id, ...) fails here,
+		 * it's because obj_id is not a user address. Hence, it is
+		 * now safe to assume that obj_id is in kernel address space.
```

Comment:
It looks a bit fragile. Only `tee_pobj_get()` calls vm_check_access_rights() to check `obj_id`. And are we sure that future caller of  `tee_pobj_get()` will have checked that?